languages/go/polar.go

```diff
@@ -195,19 +195,28 @@ func (p Polar) registerClass(cls reflect.Type, ctor interface{}, name *string) e
 		}
 	}
 
+	// get real type
+	var realType reflect.Type
+	switch c := cls.(type) {
+	case reflect.Type:
+		realType = c
+	default:
+		realType = reflect.TypeOf(cls)
+	}
+
 	// Get class name
 	var className string
 	if name == nil {
-		className = cls.Name()
+		className = realType.Name()
 	} else {
 		className = *name
 	}
 
-	err := p.host.CacheClass(cls, className, constructor)
+	err := p.host.CacheClass(realType, className, constructor)
 	if err != nil {
 		return err
 	}
-	newVal := reflect.New(cls)
+	newVal := reflect.New(realType)
```

Comment:
In case an instance (as opposed to `reflect.Type`) is provided, maybe it can be used here, instead of creating a new one?

Reply:
I'm not sure if that would work, I believe this has to be a zero value so safer to leave it like it is for now.

Reply:
Sure, I do not have a deep understanding of what is happening with the instance value down the stack and how it is used. That is why I did not change it, just left the comment, so I am completely fine with not changing this at the moment. 

I will resolve the comment.